slangpy/core/utils.py

```diff
@@ -71,6 +71,51 @@ def create_device(
     return device
 
 
+def create_torch_device(
+    type: DeviceType = DeviceType.automatic,
+    torch_device: Any = None,
+    enable_debug_layers: bool = False,
+    include_paths: Sequence[Union[str, PathLike[str]]] = [],
+    enable_print: bool = False,
+    enable_hot_reload: bool = True,
+    enable_compilation_reports: bool = False,
+):
+    """
+    Helper to create a device configured properly for PyTorch integration. If device type is CUDA,
+    slangpy will attempt to directly share the CUDA context with PyTorch. This is the recommended
+    way of using SlangPy with PyTorch.
+
+    If device type is not CUDA (eg d3d12, vulkan), this will create a device with cuda interop enabled,
+    and rely on shared memory + semaphores to syncronize between SlangPy and PyTorch. This approach
+    works, and is valuable if access to graphics features (such as a rasterizer) is critical, but hardware
+    context switching and memcpys are expensive, resulting in substantially worse performance.
+    """
+
+    # Import and init torch
+    import torch
+
+    torch.cuda.init()
+
+    # Use current device if not specified
+    if torch_device is None:
+        torch_device = torch.cuda.current_device()
+
+    # Switch to the correct device then read cuda context
+    with torch.device(torch_device):
+        handles = get_cuda_current_context_native_handles()
```

Comment:
It looks like this will throw (`No current CUDA context found.`) if the torch device has not been used yet. Minimal repro:
```
torch_device = torch.device('cuda:0')
torch.set_default_device(torch_device)
device = create_torch_device(spy.DeviceType.d3d12, torch_device)
```

It appears torch does not set the cuda context until you do something with it. In the code above, you can fix it by e.g. adding `torch.empty((1, ), device=torch_device)` before the `create_torch_device` call.

Reply:
After taking a quick look I haven't seen an obvious method in `torch.cuda` for setting the CUDA context. Weirdly,  `torch.cuda.init()` does not seem to do it.

Unfortunately the common practice of creating all devices up front before doing any work will trigger this issue. For safety, you could add a minimal torch cuda call like e.g. creating an empty 1x1 cuda tensor inside `create_torch_device`.